src/librustdoc/html/render/cache.rs

```diff
@@ -78,7 +78,7 @@ crate fn build_index(krate: &clean::Crate, cache: &mut Cache) -> String {
                 desc: item.doc_value().map_or_else(String::new, |s| short_markdown_summary(&s)),
                 parent: Some(did),
                 parent_idx: None,
-                search_type: get_index_search_type(&item, None),
+                search_type: get_index_search_type(&item, Some(cache)),
```

Comment:
Just to make sure I understand what is happening here: This adds primitives to the cache, because now in an earlier stage they are being properly added to the cache via the new `insert` method? And previously this wasn't used because... Cache wasn't working right on them, I'm guessing?

Reply:
Exactly. To be more precise: the cache here is needed to get the `DefId` of primitive types. If we don't have primitives, we don't need cache (and before that we didn't get the primitives in the cache). So you got it exactly rght. :)